scrapers/scrape_common.py

```diff
@@ -190,6 +190,76 @@ def header():
         return 'DistrictId,District,Canton,Date,Week,Year,Population,TotalConfCases,NewConfCases,TotalDeaths,NewDeaths,SourceUrl'
 
 
+class TestData:
+    __initialized = False
+    SEPARATOR = ','
+
+    def __init__(self, canton=None, url=None):
+        self.start_date = None
+        self.end_date = None
+        self.week = None
+        self.year = None
+        self.canton = canton
+        self.positive_tests = None
+        self.negative_tests = None
+        self.positivity_rate = None
+        self.url = url
+        self.__initialized = True
+
+    def __setattr__(self, key, value):
+        if self.__initialized and not hasattr(self, key):
+            raise TypeError(f'unknown key: {key}')
+        object.__setattr__(self, key, value)
+
+    def __str__(self):
+        res = []
+        res.append(self.canton)
+        res.append('' if self.start_date is None else str(self.start_date))
+        res.append('' if self.end_date is None else str(self.end_date))
+        res.append('' if self.week is None else str(self.week))
+        res.append('' if self.year is None else str(self.year))
+        res.append('' if self.positive_tests is None else str(self.positive_tests))
+        res.append('' if self.negative_tests is None else str(self.negative_tests))
+        res.append('' if self.positivity_rate is None else str(self.positivity_rate))
+        res.append(self.url)
+        return DistrictData.SEPARATOR.join(res)
+
+    def __bool__(self):
+        attributes = [
+            self.positive_tests,
+            self.negative_tests,
+            self.positivity_rate,
+        ]
+        return any(v is not None for v in attributes)
+
+    @staticmethod
+    def __get_int_item(item):
+        return int(item) if represents_int(item) else None
+
+    @staticmethod
+    def __get_float_item(item):
+        return float(item) if represents_float(item) else None
+
+    def parse(self, data):
+        items = data.split(DistrictData.SEPARATOR)
```

Comment:
Dito

Reply:
good point, should be addressed in 73ce2c6